src/libraries/System.Runtime/tests/System/GCTests.cs

```diff
@@ -892,5 +894,176 @@ long CallGetTotalAllocatedBytes(long previous)
                 previous = CallGetTotalAllocatedBytes(previous);
             }
         }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/33583", TestRuntimes.Mono)]
```

Comment:
These look like very expensive stress tests. Should they be outer loop?

Reply:
They seem fairly fast locally, but yes, they should.

I'd like to keep `AllocateArrayCheckPinning` in the inner loop though - it is a very effective test for most that may go wrong here.